src/metabase/search/scoring.clj

```diff
@@ -271,31 +275,51 @@
    [result]
    (weights-and-scores result))
 
+(defn- compute-normalized-score [scores]
+  (let [weight-sum (reduce + (map #(or (:weight %) 0) scores))
+        score-sum (reduce
+                   (fn [acc {:keys [weight score] :or {weight 0 score 0}}] (+ acc (* score weight)))
+                   0
+                   scores)]
+    (if (zero? weight-sum)
+      0
+      (/ score-sum weight-sum))))
+
+(defn force-weight [scores total]
+  (let [total-found (reduce + (map :weight scores))]
+    (mapv #(update % :weight (fn [weight]
+                               (if (contains? #{nil 0} total-found)
+                                 0
+                                 (* total (/ weight total-found))))) scores)))
+
 (defn score-and-result
-  "Returns a map with the `:score` and `:result`."
-  ([raw-search-string result]
-   (let [text-match (text-score-with-match raw-search-string result)
-         text-score {:score  (:score text-match)
-                     :weight 10
-                     :name   "text score"}
-         scores (conj (score-result result) text-score)]
-     ;; Searches with a blank search string mean "show me everything, ranked";
-     ;; see https://github.com/metabase/metabase/pull/15604 for archived search.
-     ;; If the search string is non-blank, results with no text match have a score of zero.
-     (if (or (str/blank? raw-search-string)
-             (pos? (:score text-match)))
-       {:score  (/ (reduce + (map (fn [{:keys [weight score]}] (* weight score)) scores))
-                   (reduce + (map :weight scores)))
-        :result (serialize result text-match scores)}
-       {:score 0}))))
+  "Returns a map with the normalized, combined score from relevant-scores as `:score` and `:result`."
+  [raw-search-string result]
+  (let [text-matches     (force-weight (text-scores-with-match raw-search-string result) 10)
```

Comment:
To disrupt less testing, we keep the text-matches at weight 10. We just return all of them, so we can get a clearer picture of the "why" behind result sorting.